src/ci/pgo-windows.sh

```diff
@@ -0,0 +1,134 @@
+#!/bin/bash
+
+set -euxo pipefail
+
+ci_dir=`cd $(dirname $0) && pwd`
+source "$ci_dir/shared.sh"
+
+CHECKOUT=$(pwd)
+
+DOWNLOADED_LLVM=$CHECKOUT/citools/clang-rust
+
+BUILD=$CHECKOUT/build/$PGO_HOST
+RUSTC_STAGE_0=$BUILD/stage0/bin/rustc.exe
+CARGO_STAGE_0=$BUILD/stage0/bin/cargo.exe
+RUSTC_STAGE_2=$BUILD/stage2/bin/rustc.exe
+
+# Make sure we have a temporary PGO work folder
+PGO_TMP=$CHECKOUT/build/tmp-pgo
+mkdir -p $PGO_TMP
+
+RUSTC_PERF=$PGO_TMP/rustc-perf
+
+# Compile several crates to gather execution PGO profiles.
+# Arg0 => profiles (Debug, Opt)
+# Arg1 => scenarios (Full, IncrFull, All)
+# Arg2 => crates (syn, cargo, ...)
+gather_profiles () {
+  cd $CHECKOUT
+
+  # Compile libcore, both in opt-level=0 and opt-level=3
+  RUSTC_BOOTSTRAP=1 $RUSTC_STAGE_2 \
+      --edition=2021 --crate-type=lib ./library/core/src/lib.rs \
+      --out-dir $PGO_TMP
+
+  RUSTC_BOOTSTRAP=1 $RUSTC_STAGE_2 \
+      --edition=2021 --crate-type=lib -Copt-level=3 ./library/core/src/lib.rs \
+      --out-dir $PGO_TMP
+
+  cd $RUSTC_PERF
+
+  # Run rustc-perf benchmarks
+  # Benchmark using profile_local with eprintln, which essentially just means
+  # don't actually benchmark -- just make sure we run rustc a bunch of times.
+  RUST_LOG=collector=debug \
+  RUSTC=$RUSTC_STAGE_0 \
+  RUSTC_BOOTSTRAP=1 \
+  $CARGO_STAGE_0 run -p collector --bin collector -- \
+      profile_local eprintln \
+      $RUSTC_STAGE_2 \
+      --id Test \
+      --profiles $1 \
+      --cargo $CARGO_STAGE_0 \
+      --scenarios $2 \
+      --include $3
+
+  cd $CHECKOUT
+}
+
+# We collect LLVM profiling information and rustc profiling information in
+# separate phases. This increases build time -- though not by a huge amount --
+# but prevents any problems from arising due to different profiling runtimes
+# being simultaneously linked in.
+
+# On windows, when building LLVM for PGO, for some reason, the profiler runtime library is not
+# linked in automatically so we do it manually.
+profiler_rt=$(find $DOWNLOADED_LLVM/lib/clang/ -name "clang_rt.profile-x86_64.lib")
+RUSTFLAGS="-Clink-arg=${profiler_rt}" \
+python ./x.py build --target=$PGO_HOST --host=$PGO_HOST \
+    --stage 2 library/std \
+    --llvm-profile-generate
```

Comment:
@michaelwoerister this is the most incomprehensible thing I've had to work around (just to get something working, I don't propose we land this as-is).

Similarly to building lld needing the profiler runtime, linker errors also happen while linking rustc-driver ([example output of the error](https://gist.github.com/lqd/258a608aba66c325c00e6c3382542bfa#file-linker-error-txt)) but not while building LLVM, do you think it's something we can fix by changing some cmake flags ? (seems unlikely, since LLVM was built with the flags we see in their cmake setup, [for example, see here](https://gist.github.com/lqd/258a608aba66c325c00e6c3382542bfa#file-llvm-cmake-txt))